### PR TITLE
 feat(23.10): add archive for curl and its dependencies

### PR DESCRIPTION
**Pull request by yashashwi**

### CODE REVIEW

**Comment by yashashwi** [comment]
@cjdcordeiro gentle review reminder!

**Comment by yashashwi** [comment]
@cjdcordeiro  Do you see a point in a tool like this?

image

**Comment by cjdcordeiro** [comment]
> @cjdcordeiro Do you see a point in a tool like this?
> 
> image

this is interesting @yashashwi :)

We'll be working, in the future, on expanding chisel to have additional tooling to help both the inspection and creation of slices. The slice inspection and discoverability features are already being proposed in https://github.com/canonical/chisel/pull/99 and https://github.com/canonical/chisel/pull/101.

The assisted creation of slices shall come later, but it is extremely refreshing to see the community coming up with their own solutions :)

File: slices/libssh2-1.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+package: libssh2-1
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libgcrypt20_libs
+      - zlib1g_libs
+    contents:
+      /usr/lib/*-linux-*/libssh2.so*:
```
Comment (cjdcordeiro) [inline slices/libssh2-1.yaml:15]:
```suggestion
      /usr/lib/*-linux-*/libssh2.so.1*:
```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
Thanks for the PR!

Just a few nitpicks and a leftover SDF to fix

File: slices/libldap-2.5-0.yaml
[the remark this answers is not in the document]
Comment (yashashwi) [reply slices/libldap-2.5-0.yaml:1]: removed

**Comment by yashashwi** [comment]
@cjdcordeiro  Fixed review comments.

**Comment by yashashwi** [comment]
@cjdcordeiro  review pending.

**Comment by yashashwi** [comment]
@rebornplusplus  could you please review this.

File: slices/libssh2-1.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# port forwarding, X11 forwarding; password, key-based and
+# keyboard-interactive authentication.
+
+package: libssh2-1
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libgcrypt20_libs
```
Comment (rebornplusplus) [inline slices/libssh2-1.yaml:12]:
```suggestion
      - libssl3_libs
```

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
Only one dependency seems to be wrong. Everything else looks very good. Thank you!

**Review by rebornplusplus — APPROVED** [review]
LGTM, thank you!

**Comment by yashashwi** [comment]
@cjdcordeiro  review pending.

**Comment by cjdcordeiro** [comment]
Hi @yashashwi 

apologies for the delay but we've been working on some high priority CI checks that are a must for all the pending PRs.
In the meantime, I've re-run the CLA check and it is still failing. It appears there are multiple authors in your commits, including a local user

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
failing CLA check

**Comment by yashashwi** [comment]
> failing CLA check

Yeah some of the initial commits have been as the root user. It seems I have to recommit all of it and re raise the PR.

Np! I understand you are occupied.

**Comment by linostar** [comment]
Closed in favor of #186

**Pull request closed without merging**